mod/consensus/pkg/cometbft/service/abci.go

```diff
@@ -67,98 +70,96 @@ func (s *Service[LoggerT]) InitChain(
 	)
 
 	// Set the initial height, which will be used to determine if we are
-	// proposing
-	// or processing the first block or not.
+	// proposing or processing the first block or not.
 	s.initialHeight = req.InitialHeight
-	if s.initialHeight == 0 { // If initial height is 0, set it to 1
+	if s.initialHeight == 0 {
 		s.initialHeight = 1
 	}
 
 	// if req.InitialHeight is > 1, then we set the initial version on all
 	// stores
 	if req.InitialHeight > 1 {
-		initHeader.Height = req.InitialHeight
```

Comment:
this is not really needed since Height will be overwritten in defer and initHeader is only read in defer.